crates/containerd-shim-wasm/src/sys/unix/container/instance.rs

```diff
@@ -78,8 +79,7 @@ impl<E: Engine> SandboxInstance for Instance<E> {
         // make sure we have an exit code by the time we finish (even if there's a panic)
         let guard = self.exit_code.set_guard_with(|| (137, Utc::now()));
 
-        let container_root = get_instance_root(&self.rootdir, &self.id)?;
-        let mut container = Container::load(container_root)?;
+        let mut container = self.container.write().unwrap();
```

Comment:
Is there a way to remove the usage of `unwrap()`?

Reply:
The unwrap is there due to rust's mutex poisoning.
Mutex poisoning happens when a thread panics while holding a lock on a mutex.
Since (in release) we build with `panic = "abort"` the unwrap should never happen.
In any case, if a thread panics, while holding this lock, I don't think we want to attempt recovering.

Do you think a comment about that in the code would be helpful?

Reply:
>Do you think a comment about that in the code would be helpful?

+1

Reply:
> Do you think a comment about that in the code would be helpful?

I prefer `.expect("Lock was poisoned")` or something like this

Reply:
... or use `parking_lot::RwLock` instead, if you feeling fancy :)

Reply:
I think I'll go with the `expect` option to avoid adding a dependency on `parking_lot`.
Also, panic-ing in the case of a mutex poisoning seems like a better idea than just ignoring it.